build/Makefile

```
@@ -0,0 +1,223 @@
+	$(MAKE) -C $(OSCAL_CORE_DIR) dependencies
+
+.PHONY: dependencies
+dependencies: $(JQ_PATH) $(XMLLINT_PATH) $(YQ_PATH) $(NPM_PKGS_DIR) ## Install needed jq and yq binaries, and download needed downstream dependencies
+
+# By default we install xmllint with operating system package manager, so
+# to be sensible, we will not uninstall or delete it even with the package
+# manager and reduce the amount of friction.
+.PHONY: clean-dependencies
+clean-dependencies: ## Clean binary dependencies for repo
+	rm -f $(JQ_INSTALL_PATH) $(YQ_INSTALL_PATH)
+
+#
+# OSCAL Core
+#
+
+.PHONY: build-core-artifacts
+build-core-artifacts: ## Build core OSCAL artifacts to convert content examples
+	$(MAKE) -C $(OSCAL_CORE_DIR) artifacts
+
+.PHONY: clean-core-artifacts
+clean-core-artifacts: ## Clean core OSCAL artifacts to convert content examples
+	@echo Cleaning OSCAL core artifacts
+	$(MAKE) -C $(OSCAL_CORE_DIR) clean
+
+# The directory all content is sourced from
+SRC_DIR:=../src
+# The directory all content is written to (hint: override this to ..?)
+GEN_CONTENT_DIR:=generated
+
+#
+# Readmes
+#
+
+SRC_READMES:=$(shell find $(SRC_DIR) -iname 'README.md')
+GEN_READMES:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_READMES))
+
+.PHONY: copy-readmes
+copy-readmes: $(GEN_READMES) ## Copy README files to release location
+
+# $(@D): The directory part of the file name of the target, with the
+# trailing slash removed.
+# https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html
+$(GEN_CONTENT_DIR)/%.md: $(SRC_DIR)/%.md
+	@mkdir -p $(@D)
+	cp $(SRC_DIR)/$*.md $(GEN_CONTENT_DIR)/$*.md
+
+.PHONY: clean-readmes
+clean-readmes: ## Clean generated README files
+	@echo Cleaning README content
+	rm -f $(GEN_READMES)
+
+#
+# XML Content
+#
+
+# The source xml content to use for all generated files
+SRC_FILES:=$(shell find $(SRC_DIR) -name '*.xml')
+GEN_XML_COPIED:=$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_FILES))
+SRC_XML_PROFILES:=$(shell find $(SRC_DIR) -name '*profile.xml')
+GEN_XML_RESOLVED_CATALOGS:=$(subst _profile.xml,-resolved-profile_catalog.xml,$(patsubst $(SRC_DIR)/%,$(GEN_CONTENT_DIR)/%,$(SRC_XML_PROFILES)))
+
+# All XML content generated by this configuration
+GEN_XML_FILES:=$(GEN_XML_COPIED) $(GEN_XML_RESOLVED_CATALOGS)
+
+.PHONY: copy-xml-content
+copy-xml-content: $(GEN_XML_COPIED) ## Copy OSCAL XML files to release location
+
+$(GEN_CONTENT_DIR)/%.xml: $(SRC_DIR)/%.xml
+	@mkdir -p $(@D)
+	cp $(SRC_DIR)/$*.xml $(GEN_CONTENT_DIR)/$*.xml
+
+.PHONY: resolve-xml-profiles
+resolve-xml-profiles: $(GEN_XML_RESOLVED_CATALOGS) ## Resolve OSCAL XML profiles for custom catalogs
+
+PROFILE_RESOLVER_RUNNER:=oscal/src/utils/resolver-pipeline/oscal-profile-resolve.sh
+PROFILE_RESOLVER_ARGS:="uuid-method='random-xslt'"
+
+# Do not save a backup of the in-place edited file
+SED_FLAGS:=-i
+
+ifeq ($(shell uname -s),Darwin)
+	SED_FLAGS:=-i ''
+endif
+
+$(GEN_CONTENT_DIR)/%-resolved-profile_catalog.xml: $(SRC_DIR)/%_profile.xml $(GEN_XML_COPIED)
```

NB: Small self-nitpick, I'd remove this extra target here. I tested and it all works without it.
```suggestion
$(GEN_CONTENT_DIR)/%-resolved-profile_catalog.xml: $(SRC_DIR)/%_profile.xml
```